invenio/base/scripts/apache.py

```diff
@@ -17,194 +17,414 @@
-        ssl_crt_path = CFG_PREFIX + '/etc/apache/ssl/server.crt'
-        ssl_key_path = CFG_PREFIX + '/etc/apache/ssl/server.key'
-        vhost_ip_address_needed = False
-        wsgi_socket_directive_needed = False
-        # Debian:
-        if os.path.exists(os.path.sep + 'etc' + os.path.sep + 'debian_version'):
-            listen_directive_needed = False
-            ssl_pem_directive_needed = True
-            ssl_pem_path = '/etc/apache2/ssl/apache.pem'
-            ssl_crt_path = '/etc/apache2/ssl/server.crt'
-            ssl_key_path = '/etc/apache2/ssl/server.key'
-        # RHEL/SLC:
-        if os.path.exists(os.path.sep + 'etc' + os.path.sep + 'redhat-release'):
-            listen_directive_needed = False
-            ssl_crt_path = '/etc/pki/tls/certs/localhost.crt'
-            ssl_key_path = '/etc/pki/tls/private/localhost.key'
-            vhost_ip_address_needed = True
-            wsgi_socket_directive_needed = True
-        # maybe we are using non-standard ports?
-        vhost_site_url = conf.get('CFG_SITE_URL').replace("http://", "")
-        if vhost_site_url.startswith("https://"):
-            ## The installation is configured to require HTTPS for any connection
-            vhost_site_url = vhost_site_url.replace("https://", "")
-        vhost_site_url_port = '80'
-        vhost_site_secure_url = conf.get('CFG_SITE_SECURE_URL').replace("https://", "").replace("http://", "")
-        vhost_site_secure_url_port = '443'
-        if ':' in vhost_site_url:
-            vhost_site_url, vhost_site_url_port = vhost_site_url.split(':', 1)
-        if ':' in vhost_site_secure_url:
-            vhost_site_secure_url, vhost_site_secure_url_port = vhost_site_secure_url.split(':', 1)
-        if vhost_site_url_port != '80' or vhost_site_secure_url_port != '443':
-            listen_directive_needed = True
-
-        static_root = current_app.config['COLLECT_STATIC_ROOT']
-        if not os.path.exists(static_root):
-            os.mkdir(static_root)
-
-        def prepare_alias(filename):
-            if os.path.isdir(os.path.join(static_root, filename)):
-                return '/%s/' % (filename, )
-            return '/%s' % (filename, )
-
-        aliases = map(prepare_alias, os.listdir(static_root))
-
-        apc1 = {'vhost_site_url_port': vhost_site_url_port,
-                'servername': vhost_site_url,
-                'serveralias': vhost_site_url.split('.')[0],
-                'vhost_ip_address': vhost_ip_address_needed and
-                _detect_ip_address() or '*',
-                'wsgi_socket_directive_needed': wsgi_socket_directive_needed,
-                'listen_directive_needed': listen_directive_needed,
-                'aliases': aliases,
+    class ApacheConfig(object):
+
+        """HTTP configuration resolver."""
+
+        def __init__(self):
+            """Initialize easily resolvable values."""
+            # Context
+            self._known_cert_directives = (
```

Comment:
IMHO `_known_cert_directives` and `_candidate_paths` would look better as class attributes  `KNOWN_CERT_DIRECTIVES` and `CANDIDATE_PATHS`.